repos/system_upgrade/el7toel8/actors/bindupdate/libraries/updates.py

```diff
@@ -0,0 +1,86 @@
+from leapp.libraries.common import isccfg
+from leapp.libraries.stdlib import run
+
+# Callback for walk function
+callbacks = {
+    'dnssec-lookaside': isccfg.ModifyState.callback_comment_out,
+}
+
+
+def paths_from_issues(issues):
+    """ Extract paths from list of BindConfigIssuesModel """
+    paths = []
+    for issue in issues:
+        paths.append(issue.path)
+    return paths
+
+
+def parser_file(parser, path):
+    for cfg in parser.FILES_TO_CHECK:
+        if cfg.path == path:
+            return cfg
+    return None
+
+
+def debug_log(log, text):
+    if log is not None:
+        log.debug(text)
+
+
+def make_backup(path, backup_suffix='.leapp'):
+    """ Make backup of a file before modification """
+    backup_path = path + backup_suffix
+    run(['cp', '--preserve=all', path, backup_path])
+
+
+def update_section(parser, section):
+    """ Modify one section
+    :ptype section: ConfigSection
+    """
+    state = isccfg.ModifyState()
+    parser.walk(section, callbacks, state)
+    state.finish(section)
+    return state.content()
+
+
+def update_config(parser, cfg):
+    """ Modify contents of file accoriding to rules
+    :ptype cfg: ConfigFile
+    :returns str: Modified config contents
+    """
+    return update_section(parser, cfg.root_section())
+
+
+def update_file(parser, path, log=None, write=True):
```

Comment:
what is the purpose of having a log as an argument?

Reply:
typical way of logging within this repo is to use api.current_logger() logger obj. It does some logger naming magic for you.

Reply:
okay, its purpose was to have some way to log. logger member is not available in implementation detail. api.current_logger() should work instead.